src/vulkan/vk-device.cpp

```
@@ -2110,6 +2112,61 @@ void DeviceImpl::_labelObject(uint64_t object, VkObjectType objectType, const ch
+        usage |= VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT;
+    if (is_set(desc.usage, BufferUsage::ShaderResource) &&
+        device->m_api.m_extendedFeatures.accelerationStructureFeatures.accelerationStructure)
+    {
+        usage |= VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_BIT_KHR;
+    }
+    if (hasInitData)
+        usage |= VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+    return usage;
+}
+
+Result DeviceImpl::getBufferMemoryRequirements(const BufferDesc& desc_, ResourceMemoryRequirements* outRequirements)
```

Cppcheck (2.21.0)

[style] The function 'getBufferMemoryRequirements' is never used.

(unusedFunction)